### PR TITLE
fix bug in tyche test rebasing incorrectly

### PR DESCRIPTION
**Pull request by Paul4912**
Motivation
Tests broken due to bug with rebasing incorrectly

Changes
Fixed it by making it so initial deposit doesn't call rebase and set the epoch.distribute to the whole circulating supply of Ohm.

### CODE REVIEW

File: test/tyche/YieldDirector.test.ts [hunk 1]
```diff
@@ -47,12 +40,16 @@ describe.only("YieldDirector", async () => {
...
     // Mine block and rebase. Returns the new index.
     const triggerRebase = async () => {
-        mineBlock();
+        await timeout(28800) // 8 hours per epoch
         await staking.rebase();
-
         return await sOhm.index();
     };
 
+    const timeout = async (seconds: number): Promise<void> => {
```
Comment (lolchocotaco) [inline test/tyche/YieldDirector.test.ts:48]: Can you rename this method to be still something related to mining blocks. i got it confused with `setTimeout`

File: test/tyche/YieldDirector.test.ts [hunk 2]
```diff
@@ -112,13 +109,15 @@ describe.only("YieldDirector", async () => {
...
         sOhm = await sOhmFactory.deploy();
         gOhm = await gOhmFactory.deploy(sOhm.address, sOhm.address); // Call migrate immediately
+        const blockNumBefore = await ethers.provider.getBlockNumber();
+        const blockBefore = await ethers.provider.getBlock(blockNumBefore);
         staking = await stakingFactory.deploy(
             ohm.address,
             sOhm.address,
             gOhm.address,
-            "10",
+            "28800", // 1 epoch = 8 hours
```
Comment (lolchocotaco) [inline test/tyche/YieldDirector.test.ts:118]: interesting, why did the 10 seconds not work?

**Review by lolchocotaco — APPROVED** [review]
Some comments, do all the tests pass?

**Comment by Paul4912** [comment]
> Some comments, do all the tests pass?

they do now. somehow CI is having trouble with an OZ version.

File: test/tyche/YieldDirector.test.ts (hunk 2 above)
Comment (Paul4912) [reply test/tyche/YieldDirector.test.ts:118]: yea 10 seconds is too short by time deposit happens its ready to rebase and does that thing that stuffs up the distribute.

**Comment by lolchocotaco** [comment]
LETS RIDE!

**Pull request merged**